echopype/echodata/echodata.py

```diff
@@ -0,0 +1,57 @@
+from pathlib import Path
+import xarray as xr
+from zarr.errors import GroupNotFoundError
+
+
+def open_converted(converted_raw_path, storage_options=None):
+    return EchoData(converted_raw_path=converted_raw_path, storage_options=storage_options)
+
+
+class EchoData:
+    """Echo data model class for handling raw converted data,
+    including multiple files associated with the same data set.
+    """
+
+    def __init__(self, converted_raw_path=None, storage_options=None):
+
+        # TODO: consider if should open datasets in init
+        #  or within each function call when echodata is used. Need to benchmark.
+
+        self.top = None
+        self.beam = None
+        self.beam_power = None  # only present for EK80 containing both complex and power data, may remain empty
+        self.environment = None
+        self.vendor = None
+
+        self.storage_options = storage_options if storage_options is not None else {}
+
+        # TODO: verify if converted_raw_path is valid on either local or remote filesystem
```

Comment:
@lsetiawan : this is the place where I need your help with adding checks for remote filesystems. I put in `storage_options` as in Convert.